db/compaction/compaction_iterator.cc

```diff
@@ -841,7 +844,7 @@ void CompactionIterator::NextFromInput() {
         // iteration. If the next key is corrupt, we return before the
         // comparison, so the value of has_current_user_key does not matter.
         has_current_user_key_ = false;
-        if (compaction_ != nullptr &&
+        if (compaction_ != nullptr && !compaction_->allow_ingest_behind() &&
```

Comment:
For`ikey_.type == kTypeValuePreferredSeqno`, it also checks for `compaction_->KeyNotExistsBeyondOutputLevel()`. 

Do you think it will encounter the same problem as deletion if the ikey_ is swapped with a lower seqno like 0 and collide with the later ingested-behind data?